### ISSUE TITLE
Check undocumented exceptions from derived classes or interface implementations

### ISSUE THREAD

**Issue by ronnygunawan** [tetractic/ExceptionAnalyzers#7]
Exceptions from interface implementations:

```cs
public interface IFoo
{
    void Go();
//  ~~~~~~~~~~
//  ExXXXX: 'Go()' may throw undocumented exception: OperationCanceledException, KeyNotFoundException
}

public class Bar : IFoo
{
    /// <exception cref="OperationCanceledException"></exception>
    public void Go() { }
}

public class Baz : IFoo
{
    /// <exception cref="KeyNotFoundException"></exception>
    public void Go() { }
}
```

Exceptions from abstract method overrides:

```cs
public abstract class Foo
{
    public abstract void Go();
//  ~~~~~~~~~~~~~~~~~~~~~~~~~~
//  ExXXXX: 'Go()' may throw undocumented exception: OperationCanceledException, KeyNotFoundException
}

public class Bar : Foo
{
    /// <exception cref="OperationCanceledException"></exception>
    public override void Go() { }
}

public class Baz : Foo
{
    /// <exception cref="KeyNotFoundException"></exception>
    public override void Go() { }
}
```

**Comment by carlreinke** [comment]
There is an analyzer for this, but it works the other way around – the warning is reported on the implementing/overriding member.

That analyzer isn't reporting a warning because it assumes that a supertype member with no documentation comments means "don't know if it might throw exceptions" rather than "doesn't throw exceptions".  The reason for that assumption is that if you implement/override a member declared in an undocumented third-party library then it becomes difficult to fix the diagnostic (because the easy fix can't be made in your project/solution).

I'm [planning to change](https://github.com/carlreinke/ExceptionAnalyzers/commit/10625a72838db96a89a8e0b7e82bfbfa83ed249b) this in the next version so that you do get a warning if the supertype is defined in the same project (because you _can_ easily fix it in that case).

(You can make the analyzer report a warning with the current version by placing an empty documentation comment on the supertype member, but this isn't really a reasonable solution.)

```C#
using System;
using System.Collections.Generic;

public interface IFoo
{
    ///
    void Go();
}

public class Bar : IFoo
{
    /// <exception cref="OperationCanceledException"></exception>
    public void Go() { }
    //          ~~
    //          Ex0200: 'Go()' is documented as throwing exception not documented on member in 'IFoo': OperationCanceledException
}

public class Baz : IFoo
{
    /// <exception cref="KeyNotFoundException"></exception>
    public void Go() { }
    //          ~~
    //          Ex0200: 'Go()' is documented as throwing exception not documented on member in 'IFoo': KeyNotFoundException
}
```

**Comment by carlreinke** [comment]
Fixed in v1.4.1.
